openpype/tools/publisher/publish_report_viewer/window.py

```diff
@@ -306,15 +435,13 @@ def add_filepaths(self, filepaths):
             root_item.appendRows(new_items)
 
     def remove_item_by_id(self, item_id):
-        report_item = self._report_items_by_id.get(item_id)
-        if not report_item:
-            return
+        self._handler.remove_report_item(item_id)
 
-        self._handler.remove_report_items(item_id)
-        item = self._items_by_id.get(item_id)
-
-        parent = self.invisibleRootItem()
-        parent.removeRow(item.row())
+        self._report_items_by_id.pop(item_id, None)
+        item = self._items_by_id.pop(item_id)
+        if item is not None:
```

Comment:
Note that `.pop` on a `dict` will raise `KeyError` if it does not exist and no default argument is provided to the method. See [here](https://docs.python.org/3/library/stdtypes.html#dict.pop)

```suggestion
        item = self._items_by_id.pop(item_id, None)
        if item is not None:
```

Reply:
I just accidentally removed the default value when moved the code around 😕... But thanks for documentation link 🤣

Reply:
Haha, let me know when you need me to check the PR again.